vizro-core/src/vizro/integrations/kedro/_data_manager.py

```diff
@@ -2,105 +2,106 @@
 
 from importlib.metadata import version
 from pathlib import Path
-from typing import TYPE_CHECKING, Any, Optional, Union
+from typing import Any, Optional, Union
 
 from kedro.framework.session import KedroSession
 from kedro.framework.startup import bootstrap_project
 from kedro.pipeline import Pipeline
 from packaging.version import parse
 
+# find_kedro_project was made public in 1.0.0.
+if parse(version("kedro")) >= parse("1"):
+    from kedro.utils import find_kedro_project
+else:
+    from kedro.utils import _find_kedro_project as find_kedro_project
+
+from kedro.io import DataCatalog
+
 from vizro.managers._data_manager import pd_DataFrameCallable
 
-if TYPE_CHECKING:
-    from kedro.io import CatalogProtocol
 
+def _infer_project_path(project_path) -> Union[str, Path]:
+    # Follows same logic as done interally in Kedro: if project_path not explicitly specified, try to find a Kedro
+    # project above this point in the directory tree, and if that fails then use current working directory. If
+    # project_path is not valid then bootstrap_project will raise an error.
+    return project_path or find_kedro_project(Path.cwd()) or Path.cwd()
 
-def catalog_from_project(
-    project_path: Union[str, Path], env: Optional[str] = None, extra_params: Optional[dict[str, Any]] = None
-) -> CatalogProtocol:
-    """Return the Kedro Data Catalog associated to a Kedro project.
+
+def catalog_from_project(project_path: Optional[Union[str, Path]] = None, **kwargs: Any) -> DataCatalog:
+    """Fetches the Kedro Data Catalog for a Kedro project.
 
     Args:
-        project_path: Path to the Kedro project root directory.
-        env: Kedro configuration environment to be used. Defaults to "local".
-        extra_params: Optional dictionary containing extra project parameters
-            for underlying KedroContext. If specified, will update (and therefore
-            take precedence over) the parameters retrieved from the project
-            configuration.
+        project_path: Path to the Kedro project root directory. If not specified then attempts to find a Kedro project
+            in the current directory or above.
+
+    Other Args:
+        **kwargs: Keyword arguments to pass to `KedroSession.create`, for example `env`.
 
     Returns:
-         A Kedro Data Catalog.
+         Kedro Data Catalog.
 
     Examples:
         >>> from vizro.integrations import kedro as kedro_integration
         >>> catalog = kedro_integration.catalog_from_project("/path/to/kedro/project")
     """
+    # Add tests, API docs, check narrative docs, then done
+    if kwargs.get("save_on_close"):
+        # TODO: test
+        raise ValueError("`catalog_from_project` cannot run with `save_on_close=True`.")
+    # TODO: test whole funciton
```

Comment:
Should these TODOs be done as the part of this PR? 
```suggestion
    # TODO: test whole function
```